cpp/src/io/parquet/page_decode.cuh

```diff
@@ -738,7 +738,7 @@ static __device__ void update_list_offsets_for_pruned_pages(page_state_s* state)
   auto const tid               = cg::this_thread_block().thread_rank();
 
   // Iterate by depth and store offset(s) to the list location(s)
-  for (int depth = tid; depth <= max_depth; depth += block_size) {
+  for (int depth = tid; depth < max_depth; depth += block_size) {
```

Comment:
The `next_nesting_info` at L747 would be OOB and we also don't need to write offsets for the leaf level